src/test/java/com/github/bibsysdev/nva/cristin/person/CristinPersonsHandlerTest.java

```diff
@@ -0,0 +1,227 @@
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    public void handlerSetsDefaultValueForMissingOptionalLanguageParameterAndReturnOk() throws Exception {
+        InputStream input = requestWithQueryParameters(Map.of(NAME_QUERY_PARAMETER, NAME_MATTIAS));
+        handler.handleRequest(input, output, context);
+        GatewayResponse<PersonsWrapper> response = GatewayResponse.fromOutputStream(output);
+
+        assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+        assertEquals(MediaType.APPLICATION_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    public void handlerReceivesAllowOriginHeaderValueFromEnvironmentAndPutsItOnResponse() throws Exception {
+        GatewayResponse<PersonsWrapper> response = sendDefaultQuery();
+        assertEquals(ALLOW_ALL_ORIGIN, response.getHeaders().get(ApiGatewayHandler.ACCESS_CONTROL_ALLOW_ORIGIN));
+    }
+
+    @Test
+    public void handlerReturnsBadRequestWhenNameQueryParamIsEmpty() throws Exception {
+        InputStream input = requestWithQueryParameters(Map.of(NAME_QUERY_PARAMETER, EMPTY_STRING));
+        handler.handleRequest(input, output, context);
+        GatewayResponse<PersonsWrapper> response = GatewayResponse.fromOutputStream(output);
+
+        assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        assertTrue(response.getBody().contains(CristinPersonsHandler.NAME_MISSING_EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    public void handlerReturnsBadRequestWhenReceivingInvalidLanguageQueryParam() throws Exception {
+        InputStream input = requestWithQueryParameters(
+            Map.of(NAME_QUERY_PARAMETER, NAME_MATTIAS, LANGUAGE_KEY, LANGUAGE_INVALID));
+        handler.handleRequest(input, output, context);
+        GatewayResponse<PersonsWrapper> response = GatewayResponse.fromOutputStream(output);
+
+        assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        assertTrue(response.getBody().contains(LANGUAGE_INVALID_ERROR_MESSAGE));
+    }
+
+    @Test
+    public void cristinApiClientWillStillGenerateQueryPersonsUrlEvenWithoutParameters()
+        throws IOException, URISyntaxException {
+        cristinApiClientStub.generateQueryPersonsUrl(null);
+    }
+
+    @Test
+    public void readerThrowsIoExceptionWhenReadingInvalidJson() {
+        InputStream inputStream = new ByteArrayInputStream(INVALID_JSON.getBytes(StandardCharsets.UTF_8));
+        InputStreamReader reader = new InputStreamReader(inputStream);
+        Executable action = () -> CristinApiClient.fromJson(reader, CristinPerson.class);
+        assertThrows(IOException.class, action);
+    }
+
+    @BeforeEach
+    void setUp() {
```

Comment:
should this be at the top?

Reply:
some tests being public and other not was causing IntelliJ to do this, made them all package-private for now.